src/services/contracts/safeContracts.ts

```diff
@@ -80,65 +98,61 @@ export const _getMinimumMultiSendCallOnlyVersion = (safeVersion: SafeInfo['versi
   return semver.gte(safeVersion, INITIAL_CALL_ONLY_VERSION) ? safeVersion : INITIAL_CALL_ONLY_VERSION
 }
 
-export const getMultiSendCallOnlyContract = async (
-  chainId: string,
-  safeVersion: SafeInfo['version'],
-  provider: BrowserProvider,
-) => {
-  const ethAdapter = await createEthersAdapter(provider)
-  const multiSendVersion = _getMinimumMultiSendCallOnlyVersion(safeVersion)
-
-  return ethAdapter.getMultiSendCallOnlyContract({
-    singletonDeployment: getMultiSendCallOnlyContractDeployment(chainId, multiSendVersion),
-    ..._getValidatedGetContractProps(safeVersion),
-  })
-}
+export const getReadOnlyMultiSendCallOnlyContract = async (safeVersion: SafeInfo['version']) => {
+  const safeSDK = getSafeSDK()
+  if (!safeSDK) {
+    throw new Error('Safe SDK not found.')
+  }
 
-export const getReadOnlyMultiSendCallOnlyContract = async (chainId: string, safeVersion: SafeInfo['version']) => {
-  const ethAdapter = createReadOnlyEthersAdapter()
-  const multiSendVersion = _getMinimumMultiSendCallOnlyVersion(safeVersion)
+  const safeProvider = safeSDK.getSafeProvider()
 
-  return ethAdapter.getMultiSendCallOnlyContract({
-    singletonDeployment: getMultiSendCallOnlyContractDeployment(chainId, multiSendVersion),
-    ..._getValidatedGetContractProps(safeVersion),
-  })
+  return getMultiSendCallOnlyContractInstance(_getValidatedGetContractProps(safeVersion).safeVersion, safeProvider)
 }
 
 // GnosisSafeProxyFactory
 
-export const getReadOnlyProxyFactoryContract = (chainId: string, safeVersion: SafeInfo['version']) => {
-  const ethAdapter = createReadOnlyEthersAdapter()
+export const getReadOnlyProxyFactoryContract = (safeVersion: SafeInfo['version']) => {
+  const safeSDK = getSafeSDK()
```

Comment:
Isn't the SafeSDK usually `undefined` here as well (as in `getReadonlyFallbackHandlerContract`)?  
We call this function during Safe creation where we have no context of a Safe yet.

Reply:
Nice catch! It was indeed not working e.g. for the gas estimation during safe creation. I created a helper function `getSafeProvider` and use that instead which we can also mock more easily in our tests.